orchestrator/api/helpers.py

```diff
@@ -27,11 +28,30 @@
 
 from orchestrator.api.error_handling import raise_status
 from orchestrator.db import ProcessTable, ProductTable, SubscriptionTable
+from orchestrator.db.models import SubscriptionSearchView
 from orchestrator.domain.base import SubscriptionModel
 
 logger = get_logger(__name__)
 
 
+def _quote_if_kv_pair(token: str) -> str:
+    return f'"{token}"' if ":" in token else token
+
+
+def _process_text_query(q: str) -> str:
+    quote = '"'
+    if q.count(quote) % 2 == 1:
+        q += quote  # Add missing closing quote
+    lex = shlex(q)
```

Comment:
Just wondering if there's a security risk here? I never used shlex, but seems there's a quote option to make sure that you don't pass malicious strings.

Reply:
The final textquery is passed to postgres function `websearch_to_tsquery` [which](https://www.postgresql.org/docs/current/textsearch-controls.html)

> `websearch_to_tsquery` creates a tsquery value from querytext using an alternative syntax in which simple unformatted text is a valid query. Unlike plainto_tsquery and phraseto_tsquery, it also recognizes certain operators. Moreover, this function will never raise syntax errors, **which makes it possible to use raw user-supplied input for search**

So as long as these functions are never used for any other query building, I guess it's fine

Reply:
I'm just using shlex for splitting the input string into tokens, so parts between quotes `"` are a single token. I don't see a security risk here.

Reply:
The single tokens might still be 'evil' in someway, but I guess since as Mark says we handle it with websearch_to_tsquery, we should be fine.